docs/advanced-topics/optimizations.md

```diff
@@ -34,44 +34,90 @@ impl Example {
+Arguably, the largest drawback to using Yew is the long time it takes to compile Yew apps. The time 
+taken to compile a project seems to be related to the quantity of code passed to the `html!` macro. 
+This tends to not be much of an issue for smaller projects, but for larger applications it makes 
+sense to split code across multiple crates to minimize the amount of work the compiler has to do for 
+each change made to the application.
 
-You should try to make your main crate handle routing/page selection, move all commonly shared code to another crate, and then make a different crate for each page, where each page could be a different component, or just a big function that produces `Html`. In the best case scenario, you go from rebuilding all of your code on each compile to rebuilding only the main crate, and one of your page crates. In the worst case, where you edit something in the "common" crate, you will be right back to where you started: compiling all code that depends on that commonly shared crate, which is probably everything else.
+One possible approach is to make your main crate handle routing/page selection, and then make a 
+different crate for each page, where each page could be a different component, or just a big 
+function that produces `Html`. Code which is shared between the crates containing different parts of
+the application could be stored in a separate crate which is depended on throughout the project.
+In the best case scenario, you go from rebuilding all of your code on each compile to rebuilding 
+only the main crate, and one of your page crates. In the worst case, where you edit something in the 
+"common" crate, you will be right back to where you started: compiling all code that depends on that 
+commonly shared crate, which is probably everything else.
 
-If your main crate is too heavyweight, or you want to rapidly iterate on a deeply nested page \(eg. a page that renders on top of another page\), you can use an example crate to create a more simple implementation of the main page and render your work-in-progress component on top of that.
+If your main crate is too heavyweight, or you want to rapidly iterate on a deeply nested page \(eg. 
+a page that renders on top of another page\), you can use an example crate to create a simplified 
+implementation of the main page and render the component you are working on on top of that.
 
-## Build size optimization
+## Reducing binary sizes
 
 * optimize Rust code
   * `wee_alloc` \( using tiny allocator \)
   * `cargo.toml` \( defining release profile \)
 * optimize wasm code using `wasm-opt`
 
-More information about code size profiling: [rustwasm book](https://rustwasm.github.io/book/reference/code-size.html#optimizing-builds-for-code-size)
+**Note: more information about reducing binary sizes can be found in the Rust Wasm Book (a link is
+given in the "further reading" section of this page).**
```

Comment:
This time the actual link isn't that far away but I still think it would be nice to have a clickable link in the note.